Index: src/org/apache/axis/message/addressing/Constants.java
===================================================================
RCS file: /home/cvs/ws-fx/addressing/src/org/apache/axis/message/addressing/Constants.java,v
retrieving revision 1.10
diff -u -w -r1.10 Constants.java
--- src/org/apache/axis/message/addressing/Constants.java	8 Nov 2004 18:26:21 -0000	1.10
+++ src/org/apache/axis/message/addressing/Constants.java	21 Mar 2005 19:02:14 -0000
@@ -53,7 +53,7 @@
      * Field NS_URI_ADDRESSING (default)
      */
     public static final String NS_URI_ADDRESSING =
-        NS_URI_ADDRESSING_08;
+        NS_URI_ADDRESSING_04;
 
     /**
      * Field FAULT_ACTION
