Index: src/org/apache/axis/encoding/ser/BeanSerializer.java
===================================================================
RCS file: /home/cvs/ws-axis/java/src/org/apache/axis/encoding/ser/BeanSerializer.java,v
retrieving revision 1.80
diff -w -u -r1.80 BeanSerializer.java
--- src/org/apache/axis/encoding/ser/BeanSerializer.java	11 Jan 2005 23:35:48 -0000	1.80
+++ src/org/apache/axis/encoding/ser/BeanSerializer.java	21 Mar 2005 18:45:58 -0000
@@ -175,9 +175,11 @@
                         if (propValue == null) {
                             // an element cannot be null if nillable property is set to 
                             // "false" and the element cannot be omitted
+                            /*
                             if (!isNillable && !isOmittable) {
                                 throw new IOException(Messages.getMessage("nullNonNillableElement", propName));
                             }
+                            */
                             
                             // if meta data says minOccurs=0, then we can skip
                             // it if its value is null and we aren't doing SOAP
