diff -N -P -r -c mdsweb-0.1.2/mdsutility.inc mdsweb-0.1.3/mdsutility.inc
*** mdsweb-0.1.2/mdsutility.inc	Wed Dec 11 11:37:17 2002
--- mdsweb-0.1.3/mdsutility.inc	Fri Dec 13 20:08:20 2002
***************
*** 41,47 ****
          }
          else
          {
!             $last_match .= trim($line);
          }
          if (($hostname != NULL) && ($dn != NULL))
          {
--- 41,47 ----
          }
          else
          {
!             $last_match .= $line;
          }
          if (($hostname != NULL) && ($dn != NULL))
          {
***************
*** 137,149 ****
--- 137,178 ----
      }
  }
  
+ function check_or_add_duplicate(&$dup_array, $line)
+ {
+     $ret = 0;
+     reset($dup_array);
+     while(list(,$dup) = each($dup_array))
+     {
+         if (strcmp($dup,$line) == 0)
+         {
+             $ret = 1;
+             break;
+         }
+     }
+ 
+     if ($ret == 0)
+     {
+         array_push($dup_array,$line);
+     }
+     return $ret;
+ }
+ 
  function organize_host_attributes($output,&$ret_array)
  {
      $last_match = "";
  
+     $dup_array = array();
+ 
      reset($output);
      while(list(,$line) = each($output))
      {
+         $line = trim($line);
+ 
+         if (check_or_add_duplicate($dup_array,$line) == 1)
+         {
+             continue;
+         }
+ 
          if (ereg("(.*): (.*)",$line,$matches))
          {
              if (is_array($ret_array["$matches[1]"]))
***************
*** 169,175 ****
          }
          else
          {
!             $ret_array["$last_match"] .= trim($line);
          }
      }
      return $ret_array;
--- 198,204 ----
          }
          else
          {
!             $ret_array["$last_match"] .= $line;
          }
      }
      return $ret_array;
