diff -N -P -r -c mdsweb-0.1.0/mdsgisinterface.inc mdsweb-0.1.1/mdsgisinterface.inc
*** mdsweb-0.1.0/mdsgisinterface.inc	Thu Nov 14 21:29:02 2002
--- mdsweb-0.1.1/mdsgisinterface.inc	Tue Dec 10 16:48:14 2002
***************
*** 11,16 ****
--- 11,17 ----
      }
      if (strlen($dn))
      {
+         $dn = str_replace(" ","",$dn);
          $cmd .= "-b $dn ";
      }
      $cmd .= "$filter $attributes $args";
diff -N -P -r -c mdsweb-0.1.0/mdsutility.inc mdsweb-0.1.1/mdsutility.inc
*** mdsweb-0.1.0/mdsutility.inc	Thu Nov 14 21:29:02 2002
--- mdsweb-0.1.1/mdsutility.inc	Tue Dec 10 16:48:47 2002
***************
*** 41,47 ****
          }
          else
          {
!             $last_match .= $line;
          }
          if (($hostname != NULL) && ($dn != NULL))
          {
--- 41,47 ----
          }
          else
          {
!             $last_match .= trim($line);
          }
          if (($hostname != NULL) && ($dn != NULL))
          {
***************
*** 169,175 ****
          }
          else
          {
!             $ret_array["$last_match"] .= $line;
          }
      }
      return $ret_array;
--- 169,175 ----
          }
          else
          {
!             $ret_array["$last_match"] .= trim($line);
          }
      }
      return $ret_array;
***************
*** 193,199 ****
          }
          else
          {
!             $ret_array["$last_match"] .= $line;
          }
      }
      return $ret_array;
--- 193,199 ----
          }
          else
          {
!             $ret_array["$last_match"] .= trim($line);
          }
      }
      return $ret_array;
