diff -N -P -r -c mdsweb-0.1.1/index.php mdsweb-0.1.2/index.php
*** mdsweb-0.1.1/index.php	Tue Dec 10 16:47:25 2002
--- mdsweb-0.1.2/index.php	Wed Dec 11 11:37:23 2002
***************
*** 90,123 ****
      // column 2 - OS name
      print "<td bgcolor=$default_cell_color>";
      font_start($default_font_size);
!     print $cur_array["Mds-Os-name"];
      font_end();
      print "</td>\n";
  
      // column 3 - OS release
      print "<td bgcolor=$default_cell_color>";
      font_start($default_font_size);
!     print $cur_array["Mds-Os-release"];
      font_end();
      print "</td>\n";
  
      // column 4 - Node count
      print "<td bgcolor=$default_cell_color>";
      font_start($default_font_size);
!     print $cur_array["Mds-Computer-Total-nodeCount"];
      font_end();
      print "</td>\n";
      
      // column 5 - CPU count
      print "<td bgcolor=$default_cell_color>";
      font_start($default_font_size);
!     print $cur_array["Mds-Cpu-Smp-size"];
      font_end();
      print "</td>\n";
  
      // column 6 - arch/platform
!     $arch = $cur_array["Mds-Computer-isa"];
!     $platform = $cur_array["Mds-Computer-platform"];
      print "<td bgcolor=$default_cell_color>";
      font_start($default_font_size);
      print $arch . "/" . $platform;
--- 90,135 ----
      // column 2 - OS name
      print "<td bgcolor=$default_cell_color>";
      font_start($default_font_size);
!     print (strlen($cur_array["Mds-Os-name"]) ?
!            $cur_array["Mds-Os-name"] : "N/A");
      font_end();
      print "</td>\n";
  
      // column 3 - OS release
      print "<td bgcolor=$default_cell_color>";
      font_start($default_font_size);
!     print (strlen($cur_array["Mds-Os-release"]) ?
!            $cur_array["Mds-Os-release"] : "N/A");
      font_end();
      print "</td>\n";
  
      // column 4 - Node count
      print "<td bgcolor=$default_cell_color>";
      font_start($default_font_size);
!     print (strlen($cur_array["Mds-Computer-Total-nodeCount"]) ?
!            $cur_array["Mds-Computer-Total-nodeCount"] : "N/A");
      font_end();
      print "</td>\n";
      
      // column 5 - CPU count
      print "<td bgcolor=$default_cell_color>";
      font_start($default_font_size);
!     print (strlen($cur_array["Mds-Cpu-Smp-size"]) ?
!            $cur_array["Mds-Cpu-Smp-size"] : "N/A");
      font_end();
      print "</td>\n";
  
      // column 6 - arch/platform
!     $arch = "N/A";
!     $platform = "N/A";
!     if (strlen($cur_array["Mds-Computer-isa"]))
!     {
!         $arch = $cur_array["Mds-Computer-isa"];
!     }
!     if (strlen($cur_array["Mds-Computer-platform"]))
!     {
!         $platform = $cur_array["Mds-Computer-platform"];
!     }
      print "<td bgcolor=$default_cell_color>";
      font_start($default_font_size);
      print $arch . "/" . $platform;
***************
*** 127,133 ****
      // column 7 - CPU free (15 min)
      print "<td bgcolor=$default_cell_color>";
      font_start($default_font_size);
!     print $cur_array["Mds-Cpu-Total-Free-15minX100"];
      font_end();
      print "</td>\n";
  
--- 139,146 ----
      // column 7 - CPU free (15 min)
      print "<td bgcolor=$default_cell_color>";
      font_start($default_font_size);
!     print (strlen($cur_array["Mds-Cpu-Total-Free-15minX100"]) ?
!            $cur_array["Mds-Cpu-Total-Free-15minX100"] : "N/A");
      font_end();
      print "</td>\n";
  
